apps/04TowersOfHanoi.js

```diff
@@ -16,24 +16,36 @@ function printStacks() {
     console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-    // Your code here
+function movePiece(startStack, endStack) {
+  var value = stacks[startStack].pop();
+  stacks[endStack].push(value);
 
 }
 
-function isLegal() {
-    // Your code here
-
+function isLegal(startStack, endStack) {
+  if (stacks[startStack].length === 0) {
+    return false;
+  }
+    else if(stacks[endStack].length === 0){
+      return true;
+    }
+    else {
+      return stacks[startStack][stacks[startStack].length-1] < stacks[endStack][stacks[endStack].length-1];
+    }
 }
 
 function checkForWin() {
-    // Your code here
-
+    if (stacks.a.length === 4 || stacks.b.length === 4) {
+      console.log ("You Win!!");
+      return true;
+        }
+    else {
+      return false;
+    }
 }
 
 function towersOfHanoi(startStack, endStack) {
-    // Your code here
-
+    movePiece(startStack, endStack);
 }
```

Comment:
Also, you need to call **checkForWin()** inside your **towersOfHanoi(startStack, endStack)** to see if the player has won.